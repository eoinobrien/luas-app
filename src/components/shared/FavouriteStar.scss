button.favourite-star {
  background: none;
  border: none;
  margin-left: auto;

  svg {
    color: var(--secondary-text-colour);
    width: 2rem;
  }
}
