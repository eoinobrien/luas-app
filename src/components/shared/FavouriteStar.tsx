import React, { ReactElement } from 'react';
import './FavouriteStar.scss';
import { ReactComponent as Star } from './star.svg';
import { ReactComponent as StarFill } from './star-fill.svg';

interface FavouriteStarProps {
  favouriteClick: any;
  isFavourite: boolean;
  name: string;
}

const FavouriteStar: React.FC<FavouriteStarProps> = (props: FavouriteStarProps) => {
  return (
    <button
      className="favourite-star"
      aria-label={"Favourite " + props.name}
      onClick={() => props.favouriteClick()}>
      {props.isFavourite ? <StarFill style={{color: '#fdd835'}} /> : <Star />}
    </button>
  );
}

export default FavouriteStar;
