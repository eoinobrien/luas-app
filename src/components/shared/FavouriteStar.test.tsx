import React from 'react';
import renderer from 'react-test-renderer';
import FavouriteStar from './FavouriteStar';


test('renders correctly', () => {
  const favouriteClick = () => {
    console.log("click");
  }

  const tree = renderer
    .create(<FavouriteStar isFavourite={false} favouriteClick={favouriteClick} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
