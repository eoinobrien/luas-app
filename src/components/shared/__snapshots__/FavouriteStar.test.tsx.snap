// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<button
  className="favourite-star"
  onClick={[Function]}
>
  <svg>
    star.svg
  </svg>
</button>
`;

exports[`renders learn react link 2`] = `
<button
  className="favourite-star"
  onClick={[Function]}
>
  <svg>
    star.svg
  </svg>
</button>
`;
